### DISCUSSION TITLE
Promises $nextTick throw an error: tickStack.shift(...) is not a function

### DISCUSSION THREAD

**Discussion by machine2maxin** [alpinejs/alpine#2869] [1. Help] [answered]
` <script>
    Alpine.data('box', () => ({
      list: ['a','b','c','d'],
      effect() {
        this.$nextTick(() => {
          console.log(this.$el); // ok i can get element
        });
      },
      async effect() {
        await this.$nextTick();
        console.log(this.$el); // i can still get the element but it throws an error 
        //  Uncaught TypeError: tickStack.shift(...) is not a function
      }
    }));
  </script>
  <div class="box" x-data="data">
    <template x-for="item of list">
      <div x-text="item" x-effect="effect"></div>
    </template>
  </div>`
async effect function and Promises $nextTick  is not work 
normal function and callback $nextTick is work
im very confused
looking forward to answer

**Comment by SimoTod** [comment, marked as answer]
Which version of Alpine are you using? To use `await $nextTick()` without an argument, it has to be >= 3.10.0

**Reply by machine2maxin** [reply]
I updated the version to 3.10.2 and my problem was solved. Thanks for your reply~
I have one more question I hope to get an answer to.
I will encounter some errors, but these are all triggered by alpinejs. Can I locate the location of my real code so that I can debug easily, just like the source map
Finally thank you for your reply. Good luck

**Reply by SimoTod** [reply]
Usually before the error you get a warning showing the directive that triggered the error. If you right click on it, you can jump to the dom element and you can start from there.

**Reply by machine2maxin** [reply]
Thank you for your hard work~ I think I know how to solve it

**Reply by ekwoka** [reply]
For anyone else finding this. the version is 3.10.2. Unlikely anyone will be trying to upgrade to 3.10 and not just going to latest, but who knows.
